Add creating and editing alerts documentation

idavidmcdonald opened the pull request:
https://trello.com/c/cqaglFzy/496-draft-and-publish-documentation-for-how-to-create-and-edit-alerts-using-our-paas-prometheus

First attempt as guidance for teams creating and editing their prometheus alerts.

Note, there was some debate regarding the index page of the reliability eng docs which reference a monitoring platform (https://reliability-engineering.cloudapps.digital/#monitoring) whereas our service is listed as a 'metrics tool' and if this should remain the case.

It might also be worth thinking about if this guidance is easy enough for users to find from the index page.

kentsanggds committed:
Add docs on prometheus alerts with PaaS apps

kentsanggds committed:
Update README with work around for running locally

philandstuff commented on source/index.html.md.erb:
```diff
@@ -126,3 +127,4 @@ Reliability Engineering supports each GDS team's existing service levels until s
 [how to access AWS accounts]: manuals/access-aws-accounts.html
 [how to create AWS accounts]: manuals/create-aws-accounts.html
 [how AWS cross account access works]: manuals/aws-account-management.html
+[#re-prometheus-support]: https://gds.slack.com/messages/CAF5H4N4Q
```
we can use a nicer url here: https://gds.slack.com/messages/re-prometheus-support

philandstuff commented on source/manuals/alerting-with-prometheus-for-paas.html.md.erb:
```diff
@@ -0,0 +1,85 @@
+
+Use the [Prometheus dashboard][1] to experiment writing your alert as a [PromQL][5] expression.
+
+The expression should generate some metrics which could be used in the alert, so no metrics will mean that the alert will not fire.
```
It took me a while to understand what you meant by "The expression should generate some metrics" - if it's not clear to me, I don't think it will be clear to our users.

idavidmcdonald replied:
I agree this isn't clear enough. Although now I'm also questioning whether this sentence (or what it is trying to communicate) is beneficial to users. Similar information is covered at the top of the alerting rules docs - https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/

I'm going to remove this sentence for now but feel free to disagree and I can rethink.

philandstuff commented on source/manuals/alerting-with-prometheus-for-paas.html.md.erb:
```diff
@@ -0,0 +1,85 @@
+
+The expression should generate some metrics which could be used in the alert, so no metrics will mean that the alert will not fire.
+
+Your expression must contain a `job` label that points at your team's metrics so that you don't accidently include metrics from other teams.
```
Why do we need both `job` (in this paragraph) and `space` (in the next)?  Also, it's entirely possible that different teams use the same job and space names.  Maybe we should require `org`?

idavidmcdonald replied:
Good questions.

I've given this some more thought.

I think we should always require alerts to use `org` to ensure they do not use other teams metrics (as you point out, job may not be unique). PaaS ensures that `org` will be unique.

I think it is important to also suggest teams use a `space` label as we have seen users accidentally trigger alerts for staging apps (note, this solves the problem of a team differing between their own metrics as opposed to other teams metrics as mentioned above).

Will add a new commit to refine this.

philandstuff reviewed (commented):
just a few comments. Looks good for a first draft

idavidmcdonald committed:
Add nicer URLs for slack

idavidmcdonald committed:
Change requirement of job to org label

idavidmcdonald committed:
Make language stricter

idavidmcdonald committed:
Remove sentence which added little value

idavidmcdonald committed:
Add team name to example alerting rule name

philandstuff commented on source/manuals/alerting-with-prometheus-for-paas.html.md.erb:
```diff
@@ -40,7 +40,7 @@ An alerting rule file will look something like this:
 - name: Your team name
   rules:
-  - alert: RequestsExcess5xx
+  - alert: TeamName_RequestsExcess5xx
```
this is contingent on our decision in alphagov/prometheus-aws-configuration-beta#82

philandstuff replied:
(but for the moment let's go with it as written, we can update later)

philandstuff commented on source/manuals/alerting-with-prometheus-for-paas.html.md.erb:
```diff
@@ -0,0 +1,83 @@
+
+Create a pull request with your changes to your alerting rules file. Your commit should explain what the alert is and why the threshold value was picked. This is important so future team members have the context they need to confidently change the alerting rule and other teams can learn from your alerting rules.
+
+Slack the [#re-prometheus-support][7] channel with the PR so that it can be reviewed. We will try and merge and deploy your pull request as quickly as possible. We will let you know when your alerting rule is live.
```
"as quickly as possible" is nice, but can we set some expectation about how long this will take in practice? Maybe "we aim to deploy a PR within by end of next working day after it is merged"?

J-Lambo committed:
Style edits

Style edits only to remove passive voice, bring in line with GOV.UK style and improve readability.

idavidmcdonald committed:
Small style tweak

idavidmcdonald committed:
Change `job` to `team` as you may have metrics from multiple jobs

J-Lambo reviewed (commented):
+1 Looks good to me

philandstuff commented on source/manuals/alerting-with-prometheus-for-paas.html.md.erb:
```diff
@@ -0,0 +1,83 @@
+
+When deciding what alerts you want to receive, consider: 
+
+- reading the[Alerting principles][6] in the GDS Way
```
missing space: `the[Alerting principles][6]` should be `the [Alerting principles][6]`

philandstuff replied:
can probably fix this then just merge?

idavidmcdonald committed:
Fix missing space

idavidmcdonald committed:
Remove trailing spaces on sentences

idavidmcdonald commented:
Right, I think this should be good to go. If you want to leave an approving review to confirm please @philandstuff then I think we are in business!
